src/RoomsSDKAdapter.js

```diff
@@ -134,4 +153,138 @@ export default class RoomsSDKAdapter extends RoomsAdapter {
+      FETCHED_CONVERSATIONS = true;
+    }
+
+    return this.datasource.internal.conversation.listActivities({
+      conversationId,
+      limit: limit + 1,
+      lastActivityFirst: true,
+      maxDate: earliestActivityDate === null ? undefined : earliestActivityDate,
+    });
+  }
+
+  /**
+   * Returns `true` if there are more activities to load from the room of the given ID.
+   * Otherwise, it returns `false`.
+   *
+   * @param {string} ID ID of the room for which to verify activities.
+   * @returns {boolean} `true` if room has more activities to load, `false` otherwise
+   */
+  hasMoreActivities(ID) {
+    const roomObservable = this.roomObservables[ID];
+    const {
+      hasMore = true,
+    } = this.roomActivities.get(ID);
+
+    if (!hasMore) {
+      roomObservable.complete();
+    } else {
+      this.fetchPastActivities(ID);
+    }
+
+    return hasMore;
+  }
+
+  /**
+   * Fetches past activities and returns array of (id, published) objects.
+   *
+   * @param {string} ID The id of the room
+   * @returns {Subscription} Observable stream that emits activity data
+   */
+  fetchPastActivities(ID) {
+    const roomActivity = this.roomActivities.get(ID);
+    const {earliestActivityDate} = roomActivity;
+    const roomObservable = this.roomObservables[ID];
+
+    logger.debug('ROOM', 'fetchPastActivities()', ['called with', {
+      ID,
+      earliestActivityDate,
+    }]);
+
+    if (!ID) {
+      roomObservable.error(new Error('getPreviousActivities - Must provide room ID'));
+    }
+
+    return from(this.fetchActivities({
+      ID,
+      earliestActivityDate,
+    }))
+      .subscribe((data) => {
+        if (!data) {
+          return roomObservable.complete();
+        }
+        roomActivity.hasMore = data.length >= this.limit + 1;
+        const {published} = data.shift();
+        const activityIds = sortByPublished(data).map((activity) => {
+          const {id} = activity;
+
+          roomActivity.activities.set(id, activity);
+
+          return [id, activity.published];
+        });
+
+        roomActivity.earliestActivityDate = published;
+        roomActivity.activityIds[published] = activityIds.length;
+
+        this.roomActivities.set(ID, roomActivity);
+
+        return roomObservable.next(activityIds);
+      });
+  }
+
+  /**
+   * Returns an observable that emits an array of the next chunk of previous
+   * activity data of the given roomID. If `hasMoreActivities` returns false,
+   * the observable will complete.
+   * **Previous activity data must be sorted newest-to-oldest.**
+   *
+   * @param {string} ID  ID of the room for which to get activities.
+   * @param {number} limit The maximum number of activities to return
+   * @returns {external:Observable.<Room>} Observable stream that emits activity data
+   */
+  getPreviousActivities(ID, limit = 50) {
+    this.limit = limit;
+    const roomObservable = this.roomObservables[ID] || new Subject();
```

Comment:
I like how @taymoork2 [named it on his PR, adding the word "cache"](https://github.com/webex/sdk-component-adapter/pull/280/files#diff-d50aaa2398d276e5ac207ce552bc89420fcf2320576bbce3660e516aa68d26ebR42) to indicate its purpose more clearly.
There are also other room observables also stored, so I would like to see this named more specifically

```suggestion
    const activitiesObservableCache = this.roomObservables[ID] || new Subject();
```
or
```suggestion
    const pastActivities$Cache = this.roomObservables[ID] || new Subject();
```